src/ORM/DataObject.php

```diff
@@ -3724,6 +3724,51 @@ public function onAfterBuild()
         $this->extend('onAfterBuild');
     }
 
+    private function getDatabaseBackedField(string $fieldPath): ?string
+    {
+        $component = $this;
+        $fieldParts = [];
+
+        foreach ($parts = explode('.', $fieldPath ?? '') as $nextPart) {
+            if (!$component) {
+                return null;
+            }
+            $fieldParts[] = $nextPart;
+            
+            if ($component instanceof Relation || $component instanceof DataList) {
+                if ($component->hasMethod($nextPart)) {
+                    // If the next part is a method, we don't have a database-backed field.
+                    return null;
+                }
+                // The next part could either be a field, or another relation
+                $singleton = DataObject::singleton($component->dataClass());
+                if ($singleton->dbObject($nextPart)) {
```

Comment:
```suggestion
                if ($singleton->dbObject($nextPart) instanceof DBField) {
```
We should probably be strict here, the same as we are for the `dbObject()` call in the condition below.

Reply:
Updated